internal/pkg/mining/scheduler.go

```diff
@@ -51,56 +61,77 @@ func (s *timingScheduler) Start(miningCtx context.Context) (<-chan Output, *sync
 	// loop mining work
 	doneWg.Add(1)
 	go func() {
-		s.mineLoop(miningCtx, outCh, &doneWg)
+		err := s.mineLoop(miningCtx, outCh)
+		if err != nil {
+			outCh <- NewOutputErr(err)
+		}
 		doneWg.Done()
 	}()
 	s.isStarted = true
 
 	return outCh, &doneWg
 }
 
-func (s *timingScheduler) mineLoop(miningCtx context.Context, outCh chan Output, doneWg *sync.WaitGroup) {
-	// mineLoop is the main event loop for the timing scheduler.  It waits for
-	// a new epoch to start, polls the heaviest head, includes the correct number
+func (s *timingScheduler) mineLoop(ctx context.Context, outCh chan Output) error {
+	mineCh := make(chan Output, 1)
+
+	// start on epoch boundary
+	targetEpoch := s.chainClock.WaitNextEpoch(ctx)
```

Comment:
Fine for now but it looks like this is not setup to handle mining block 1 during the genesis epoch.  Probably best to catch that edge case with work to do "catchup mining"